pallets/passkey/src/lib.rs

```diff
@@ -83,10 +87,33 @@ pub mod module {
 	impl<T: Config> Pallet<T> {
 		/// proxy call
 		#[pallet::call_index(0)]
-		#[pallet::weight(T::WeightInfo::proxy())]
-		pub fn proxy(_origin: OriginFor<T>) -> DispatchResult {
-			Self::deposit_event(Event::PlaceHolderEvent);
-			Ok(())
+		#[pallet::weight({
+			let dispatch_info = payload.passkey_call.call.get_dispatch_info();
+			// TODO: calculate overhead after all validations
+			let overhead = T::WeightInfo::proxy();
+			let total = overhead.saturating_add(dispatch_info.weight);
+			(total, dispatch_info.class)
+		})]
+		pub fn proxy(
+			origin: OriginFor<T>,
+			payload: PasskeyPayload<T>,
+		) -> DispatchResultWithPostInfo {
+			ensure_none(origin)?;
+			let main_origin = T::RuntimeOrigin::from(frame_system::RawOrigin::Signed(
+				payload.passkey_call.account_id,
+			));
+			payload.passkey_call.call.dispatch(main_origin)
+		}
+	}
+
+	#[pallet::validate_unsigned]
+	impl<T: Config> ValidateUnsigned for Pallet<T> {
+		type Call = Call<T>;
+		fn validate_unsigned(
```

Comment:
Did we decided which way we're going with the design--Signed or Unsigned? I didn't see an update on that question.

Reply:
I think the assumption is that we are going with unsigned for now.